### PR TITLE
[fix]: [CDS-37657]: adding security context when sending acr token fetch task

### PR DESCRIPTION
**Pull request by markoharness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33203)

### CODE REVIEW

**Comment by markoharness** [comment]
trigger messagecheck

**Comment by markoharness** [comment]
trigger ti0

**Comment by markoharness** [comment]
trigger ti0

**Review by ivan-83 — APPROVED** [review]
> Reviewed 1 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33203)* status: 1 file(s) to review. 

___
*[`120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java` line 277 at r1](https://reviewable.io/reviews/harness/harness-core/33203#-N2FvddV-zhna9nHa9wv:-N2FvddV-zhna9nHa9ww:b-nqcywn) ([raw file](https://github.com/harness/harness-core/blob/3b9de524bb675ccf246a263232c6d5ee5c11318f/120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java#L277)):*
> ```Java
>               AmbianceUtils.getOrgIdentifier(ambiance), AmbianceUtils.getProjectIdentifier(ambiance));
> 
>           Principal principal = SecurityContextBuilder.getPrincipal();
> ```

Do we need principal for SP with secret flow?
___
*[`120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java` line 279 at r1](https://reviewable.io/reviews/harness/harness-core/33203#-N2FvkqIAZAPxGOQ-Qhc:-N2FvkqIAZAPxGOQ-Qhd:bjw0ke6) ([raw file](https://github.com/harness/harness-core/blob/3b9de524bb675ccf246a263232c6d5ee5c11318f/120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java#L279)):*
> ```Java
>           Principal principal = SecurityContextBuilder.getPrincipal();
>           if (principal == null) {
>             principal = new ServicePrincipal(NG_MANAGER.getServiceId());
> ```

I guess Service principal will work for RBAC hence Service principal has unlimited access to all service. You can confirm this PL team. 
___
*[`120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java` line 282 at r1](https://reviewable.io/reviews/harness/harness-core/33203#-N2FvWAq2py7naR2ZRL5:-N2FvWAq2py7naR2ZRL6:b-n1ue7b) ([raw file](https://github.com/harness/harness-core/blob/3b9de524bb675ccf246a263232c6d5ee5c11318f/120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java#L282)):*
> ```Java
>             SecurityContextBuilder.setContext(principal);
>           }
>           log.info(format("SecurityContext is %s service", principal.getName()));
> ```

Minor - SecurityContext principal name is
___
*[`120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java` line 304 at r1](https://reviewable.io/reviews/harness/harness-core/33203#-N2Fw9cu7pTg5I5DXgl7:-N2Fw9cu7pTg5I5DXgl8:b-7ma07p) ([raw file](https://github.com/harness/harness-core/blob/3b9de524bb675ccf246a263232c6d5ee5c11318f/120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java#L304)):*
> ```Java
>           String accessToken = format("\"%s\"", accessTokenResponse.getToken());
> 
>           if (log.isDebugEnabled()) {
> ```

Minor - I will remove this,  i am not sure should we log token due to security reasons.

**Review by ivan-83 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33203#-:-N2Fyd3O6NpnK28B31gh:bnfp4nl)* status: 1 file(s) to review.

File: 120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java [hunk 1]
```diff
@@ -251,45 +255,65 @@ private void getImageDetailsFromArtifactory(ArtifactoryArtifactOutcome artifacto
...
+          }
+
+          imageDetailsBuilder.username(ACR_SP_CERT_DOCKER_USERNAME);
+          imageDetailsBuilder.password(accessToken);
+        }
+      } else {
+        log.info("This is a Managed Identity config. No credentials necessary for image pull.");
       }
+    } catch (Exception e) {
+      throw new InvalidRequestException(e.getMessage(), e);
```
Comment (gitNavV) [inline 120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java:316]: Ask: Are we further handling/catching this Exception? (keeping in mind the readability)

**Review by ivan-83 — APPROVED** [review]
> Reviewed 1 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33203)* status: :shipit: complete! ✓

File: 120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java (hunk 1 above)
Comment (gitNavV) [inline 120-ng-manager/src/main/java/io/harness/pms/expressions/utils/ImagePullSecretUtils.java:316]: Ask: Are we further handling/catching this Exception? (keeping in mind the readability)

**Pull request merged**